core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -66,7 +100,8 @@ private[spark] object TaskDescription {
     }
   }
 
-  def encode(taskDescription: TaskDescription): ByteBuffer = {
+  @throws[TaskNotSerializableException]
+  def encode(taskDescription: TaskDescription, serializedTask: ByteBuffer): ByteBuffer = {
```

Comment:
Part of it -- although much of that commit is helpful.  I just submitted a [PR](https://github.com/witgo/spark/pull/2) to your branch with what I had in mind.